lib/github_service.rb

```diff
@@ -25,12 +39,17 @@ def delete_comments(fq_repo_name, comment_ids)
 
     # Deletes the issue comments found by the provided block, then creates new
     # issue comments from those provided.
-    def replace_comments(fq_repo_name, issue_number, new_comments)
+    def replace_comments(fq_repo_name, issue_number, new_comments, status = nil, commit_sha = nil)
       raise "no block given" unless block_given?
 
       to_delete = issue_comments(fq_repo_name, issue_number).select { |c| yield c }
       delete_comments(fq_repo_name, to_delete.map(&:id))
-      add_comments(fq_repo_name, issue_number, new_comments)
+      first_comment = add_comments(fq_repo_name, issue_number, new_comments).first
+
+      # add_status creates a commit status pointing to the first comment URL
+      unless first_comment.nil? && status.nil?
```

Comment:
@Fryguy Can it be like this?
```ruby
if first_comment && !status.nil?
  add_status(...)
end
```
The status variable contain a boolean value so your example will not work in case of status has `False` value.

Reply:
Ah ok, then yes, `first_comment && !status.nil?` is good.